File: frontend/src/types/job.ts
export type JobStatus = "PENDING" | "RUNNING" | "COMPLETED" | "FAILED";

export interface JobEventItem {
  message: string;
  created_at: string;
}

export interface JobStatusResponse {
  id: string;
  status: JobStatus;
  progress: number;
  current_stage: string | null;
  error_message: string | null;
  events: JobEventItem[];
}

File: backend/app/schemas/job.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class JobEventItem(BaseModel):
    message: str
    created_at: datetime


class JobStatusResponse(BaseModel):
    id: UUID
    status: str
    progress: int
    current_stage: str | None
    error_message: str | None
    events: list[JobEventItem] = []
